dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTeiAnalyticsTableManager.java

```diff
@@ -349,24 +340,20 @@ protected void populateTable( AnalyticsTableUpdateParams params, AnalyticsTableP
 
         removeLastComma( sql )
             .append( " from trackedentityinstance tei" )
-            .append( " left join trackedentitytype tet on tet.trackedentitytypeid = tei.trackedentitytypeid" )
-            .append( " left join programinstance pi on pi.trackedentityinstanceid = tei.trackedentityinstanceid" )
-            .append( " and tei.deleted is false" )
-            .append( " left join program p on p.programid = pi.programid and pi.deleted is false" )
-            .append( " left join programstageinstance psi on psi.programinstanceid = pi.programinstanceid" )
-            .append( " left join programstage ps on ps.programstageid = psi.programstageid" )
             .append( " left join organisationunit ou on tei.organisationunitid = ou.organisationunitid" )
             .append( " left join _orgunitstructure ous on ous.organisationunitid = ou.organisationunitid" )
-            .append(
-                " left join _organisationunitgroupsetstructure ougs on pi.organisationunitid=ougs.organisationunitid " )
-            .append(
-                " and (cast(date_trunc('month', tei.created) as date)=ougs.startdate or ougs.startdate is null) " )
```

Comment:
this seems to be something related to programInstance, so I don't see how er need it here. Maybe the condition should have been on `tei.organisationunitid` ? @maikelarabori  ?

Reply:
The topic was discussed and agreed we don't actually need that left join.